<a href="https://colab.research.google.com/github/Motaha12/projetta/blob/main/Analysedonneeproject.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [44]:
import pandas as pd
import numpy as np
import matplotlib.pylab as plt


this db = pd.read_csv(url)= implemente des classes pour la lecture et l'ecriture des donnees et depend au format choisi .
db : c' est pour l'affichage de la base de donnee . 

In [45]:
url = "https://raw.githubusercontent.com/EMSIMa/ADD3IIR/main/Projet/Datasets/amazon.csv"
db = pd.read_csv(url)

In [46]:
db

show_id     type                    title         director  \
0         s1    Movie      The Grand Seduction     Don McKellar   
1         s2    Movie     Take Care Good Night     Girish Joshi   
2         s3    Movie     Secrets of Deception      Josh Webber   
3         s4    Movie       Pink: Staying True   Sonia Anderson   
4         s5    Movie            Monster Maker     Giles Foster   
...      ...      ...                      ...              ...   
9663   s9664    Movie      Pride Of The Bowery  Joseph H. Lewis   
9664   s9665  TV Show            Planet Patrol              NaN   
9665   s9666    Movie                  Outpost     Steve Barker   
9666   s9667  TV Show  Maradona: Blessed Dream              NaN   
9667   s9668    Movie              Harry Brown    Daniel Barber   

                                                   cast         country  \
0        Brendan Gleeson, Taylor Kitsch, Gordon Pinsent          Canada   
1      Mahesh Manjrekar, Abhay Mahajan, Sachin Khedekar           India   
2     Tom Sizemore, Lorenzo Lamas, Robert LaSardo, R...   United States   
3     Interviews with: Pink, Adele, Beyoncé, Britney...   United States   
4     Harry Dean Stanton, Kieran O'Brien, George Cos...  United Kingdom   
...                                                 ...             ...   
9663                           Leo Gorcey, Bobby Jordan             NaN   
9664  DICK VOSBURGH, RONNIE STEVENS, LIBBY MORRIS, M...             NaN   
9665  Ray Stevenson, Julian Wadham, Richard Brake, M...             NaN   
9666  Esteban Recagno, Ezequiel Stremiz, Luciano Vit...             NaN   
9667  Michael Caine, Emily Mortimer, Joseph Gilgun, ...             NaN   

          date_added  release_year rating   duration                listed_in  \
0     March 30, 2021          2014    NaN    113 min            Comedy, Drama   
1     March 30, 2021          2018    13+    110 min     Drama, International   
2     March 30, 2021          2017    NaN     74 min  Action, Drama, Suspense   
3     March 30, 2021          2014    NaN     69 min              Documentary   
4     March 30, 2021          1989    NaN     45 min           Drama, Fantasy   
...              ...           ...    ...        ...                      ...   
9663             NaN          1940     7+     60 min                   Comedy   
9664             NaN          2018    13+  4 Seasons                 TV Shows   
9665             NaN          2008      R     90 min                   Action   
9666             NaN          2021  TV-MA   1 Season            Drama, Sports   
9667             NaN          2010      R    103 min  Action, Drama, Suspense   

                                            description  
0     A small fishing village must procure a local d...  
1     A Metro Family decides to fight a Cyber Crimin...  
2     After a man discovers his wife is cheating on ...  
3     Pink breaks the mold once again, bringing her ...  
4     Teenage Matt Banting wants to work with a famo...  
...                                                 ...  
9663  New York City street principles get an East Si...  
9664  This is Earth, 2100AD - and these are the adve...  
9665  In war-torn Eastern Europe, a world-weary grou...  
9666  The series tells the story of Diego Maradona, ...  
9667  Harry Brown, starring two-time Academy Award w...  

[9668 rows x 12 columns]

# Question 1: top 5 des meilleurs categories :

nlargest()= il retourne  les 5 types de contenu les plus fréquents.
value_counts= retourne combien de fois chaque type est apparru dans la base de donnee .
db["type"]= c'est la base de donnee qui prend en parametre le type "categorie".

In [9]:
top_categories = db["type"].value_counts().nlargest(5)
print(top_categories)

Movie      7814
TV Show    1854
Name: type, dtype: int64


# Question 2: les top 5 des realisateurs : 

c'est le meme fonctionnement du code precedant juste on remplace "type" par "director"

In [10]:
top_realisateurs = db["director"].value_counts().nlargest(5)
print(top_realisateurs)

Mark Knight              113
Cannis Holder             61
Moonbug Entertainment     37
Jay Chapman               34
Arthur van Merwijk        30
Name: director, dtype: int64


# Question 3 : les 10 meilleurs acteurs :

Ce code extrait les acteurs du DataFrame "db" en les séparant en fonction de la virgule suivie d'un espace dans la colonne "cast". Ensuite, il compte le nombre de fois que chaque acteur apparaît dans le DataFrame en utilisant "stack()", puis renvoie les dix acteurs les plus fréquents. Ces dix acteurs sont affichés avec le nombre de fois où ils apparaissent dans le DataFrame.

In [11]:
top_acteurs = db["cast"].str.split(", ", expand=True).stack().value_counts().nlargest(10)
print(top_acteurs)

Maggie Binkley        56
1                     35
Gene Autry            32
Nassar                30
Champion              29
Anne-Marie Newland    25
Prakash Raj           24
John Wayne            23
Roy Rogers            23
Danny Trejo           22
dtype: int64


# Question 4 : 

# Question 5: Identifier des contenus similaires en faisant correspondre des caractéristiques textuelles :

"db.duplicated(subset=['title'])" renvoie un tableau booléen qui indique pour chaque ligne du DataFrame db si elle est en doublon ou non en se basant sur la colonne "title".

In [18]:
db.duplicated(subset=['title'])

0       False
1       False
2       False
3       False
4       False
        ...  
9663    False
9664    False
9665    False
9666    False
9667    False
Length: 9668, dtype: bool

# Question 6: Netflix/Amazon/Disney se concentre-t-il davantage sur les séries télévisées que sur les films ces dernières années ?

# Question 7: Affichage de la dimension de la base de donnee:

ce code retourne le nombre des lignes et nombre de colonnes dans la base de donnee : "9668"=nombre de lignes .
         "12" = nombre de colonnes .  

In [21]:
db.shape

(9668, 12)

# Question 8: les 10 premieres lignes de la base de donnees:

In [24]:
db.head(10)

show_id   type                  title                     director  \
0      s1  Movie    The Grand Seduction                 Don McKellar   
1      s2  Movie   Take Care Good Night                 Girish Joshi   
2      s3  Movie   Secrets of Deception                  Josh Webber   
3      s4  Movie     Pink: Staying True               Sonia Anderson   
4      s5  Movie          Monster Maker                 Giles Foster   
5      s6  Movie  Living With Dinosaurs                 Paul Weiland   
6      s7  Movie              Hired Gun                  Fran Strine   
7      s8  Movie           Grease Live!  Thomas Kail, Alex Rudzinski   
8      s9  Movie        Global Meltdown                Daniel Gilboy   
9     s10  Movie         David's Mother        Robert Allan Ackerman   

                                                cast         country  \
0     Brendan Gleeson, Taylor Kitsch, Gordon Pinsent          Canada   
1   Mahesh Manjrekar, Abhay Mahajan, Sachin Khedekar           India   
2  Tom Sizemore, Lorenzo Lamas, Robert LaSardo, R...   United States   
3  Interviews with: Pink, Adele, Beyoncé, Britney...   United States   
4  Harry Dean Stanton, Kieran O'Brien, George Cos...  United Kingdom   
5  Gregory Chisholm, Juliet Stevenson, Brian Hens...  United Kingdom   
6  Alice Cooper, Liberty DeVitto, Ray Parker Jr.,...   United States   
7  Julianne Hough, Aaron Tveit, Vanessa Hudgens, ...   United States   
8  Michael Paré, Leanne Khol Young, Patrick J. Ma...          Canada   
9    Kirstie Alley, Sam Waterston, Stockard Channing   United States   

       date_added  release_year rating duration  \
0  March 30, 2021          2014    NaN  113 min   
1  March 30, 2021          2018    13+  110 min   
2  March 30, 2021          2017    NaN   74 min   
3  March 30, 2021          2014    NaN   69 min   
4  March 30, 2021          1989    NaN   45 min   
5  March 30, 2021          1989    NaN   52 min   
6  March 30, 2021          2017    NaN   98 min   
7  March 30, 2021          2016    NaN  131 min   
8  March 30, 2021          2017    NaN   87 min   
9   April 1, 2021          1994    NaN   92 min   

                           listed_in  \
0                      Comedy, Drama   
1               Drama, International   
2            Action, Drama, Suspense   
3                        Documentary   
4                     Drama, Fantasy   
5                      Fantasy, Kids   
6      Documentary, Special Interest   
7                             Comedy   
8  Action, Science Fiction, Suspense   
9                              Drama   

                                         description  
0  A small fishing village must procure a local d...  
1  A Metro Family decides to fight a Cyber Crimin...  
2  After a man discovers his wife is cheating on ...  
3  Pink breaks the mold once again, bringing her ...  
4  Teenage Matt Banting wants to work with a famo...  
5  The story unfolds in a an English seaside town...  
6  They are the "First Call, A-list" musicians, j...  
7  This honest, uncompromising comedy chronicles ...  
8  A helicopter pilot and an environmental scient...  
9  Sally Goodson is a devoted mother to her autis...

# Question 9: les 10 derniers lignes de la base de donnee:

In [27]:
db.tail(10)

show_id     type                       title  \
9658   s9659    Movie       12 Dates of Christmas   
9659   s9660    Movie  10 Things I Hate About You   
9660   s9661    Movie          The Man in the Hat   
9661   s9662  TV Show                  Terrahawks   
9662   s9663    Movie                       River   
9663   s9664    Movie         Pride Of The Bowery   
9664   s9665  TV Show               Planet Patrol   
9665   s9666    Movie                     Outpost   
9666   s9667  TV Show     Maradona: Blessed Dream   
9667   s9668    Movie                 Harry Brown   

                                 director  \
9658                                  NaN   
9659                           Gil Junger   
9660  John-Paul Davidson, Stephen Warbeck   
9661                                  NaN   
9662                           Emily Skye   
9663                      Joseph H. Lewis   
9664                                  NaN   
9665                         Steve Barker   
9666                                  NaN   
9667                        Daniel Barber   

                                                   cast country date_added  \
9658                                                NaN     NaN        NaN   
9659  Heath Ledger, Julia Stiles, Joseph Gordon-Levi...     NaN        NaN   
9660             Ciaran Hinds, Stephen Dillane, Maïwenn     NaN        NaN   
9661  DENISE BRYER, WINDSOR DAVIES, JEREMY HITCHEN, ...     NaN        NaN   
9662  Mary Cameron Rogers, Alexandra Rose, Rob Marsh...     NaN        NaN   
9663                           Leo Gorcey, Bobby Jordan     NaN        NaN   
9664  DICK VOSBURGH, RONNIE STEVENS, LIBBY MORRIS, M...     NaN        NaN   
9665  Ray Stevenson, Julian Wadham, Richard Brake, M...     NaN        NaN   
9666  Esteban Recagno, Ezequiel Stremiz, Luciano Vit...     NaN        NaN   
9667  Michael Caine, Emily Mortimer, Joseph Gilgun, ...     NaN        NaN   

      release_year rating   duration                         listed_in  \
9658          2011     PG     87 min                    Drama, Romance   
9659          1999  PG-13     97 min            Comedy, Drama, Romance   
9660          2021    13+     96 min                            Comedy   
9661          1983    ALL  3 Seasons                          TV Shows   
9662          2021    16+     93 min  Drama, Science Fiction, Suspense   
9663          1940     7+     60 min                            Comedy   
9664          2018    13+  4 Seasons                          TV Shows   
9665          2008      R     90 min                            Action   
9666          2021  TV-MA   1 Season                     Drama, Sports   
9667          2010      R    103 min           Action, Drama, Suspense   

                                            description  
9658  Setup on a Christmas Eve date, a woman must re...  
9659  On the first day at his new school, Cameron in...  
9660  The Man in the Hat journeys through France in ...  
9661  This thrilling, action-packed series saw Gerry...  
9662  River is a grounded Sci-Fi mystery Thriller, t...  
9663  New York City street principles get an East Si...  
9664  This is Earth, 2100AD - and these are the adve...  
9665  In war-torn Eastern Europe, a world-weary grou...  
9666  The series tells the story of Diego Maradona, ...  
9667  Harry Brown, starring two-time Academy Award w...

# Question 10: les titres des films et tv shows dans ma base de donnee:

ce code retourne les titres de la base donnee des le debut jusqu'a la fin .

In [41]:
db.title

0           The Grand Seduction
1          Take Care Good Night
2          Secrets of Deception
3            Pink: Staying True
4                 Monster Maker
                 ...           
9663        Pride Of The Bowery
9664              Planet Patrol
9665                    Outpost
9666    Maradona: Blessed Dream
9667                Harry Brown
Name: title, Length: 9668, dtype: object

# Question 10 : l'affichage des directeurs dans la base de donnee 

pour afficher les elements de la base de donnee il suffit de taper le nom du base de donnee utilise dans to code et ajoute "." pour l'appel de l'element voulu pour qu'il saffiche 

In [47]:
db.director

0          Don McKellar
1          Girish Joshi
2           Josh Webber
3        Sonia Anderson
4          Giles Foster
             ...       
9663    Joseph H. Lewis
9664                NaN
9665       Steve Barker
9666                NaN
9667      Daniel Barber
Name: director, Length: 9668, dtype: object